### PR TITLE
Still Looking to have a working better version of Guake

### PR DESCRIPTION
**Pull request by zeusintuivo**
This fixes issues left on Master that were merged on the latest merges. 
On of the been that it does not work on latest gnome environments. 
I attempt to bump python to 3.10
This version compiles in my 2 boxes and it works better than the "apt" or "dnf" version. 
It also have lots of more REGEX 
and it fixes the issue of clicking to open which is not working in the master version.

### CODE REVIEW

File: po/ca.po [hunk 1]
```diff
@@ -1530,80 +1530,6 @@ msgstr "Voleu restablir els colors personalitzats d'aquesta pestanya?"
 msgid "New tabs appear after the current tab"
 msgstr "Les noves pestanyes apareixen després de la pestanya actual"
 
-#: guake/prefs.py:186
```
Comment (Davidy22) [inline po/ca.po:1533]: Why is all this deleted? Have you not updated your repo to master yet?

File: AUTHORS [hunk 2]
```diff
@@ -0,0 +1,304 @@
+A S Alam <apreet.alam@gmail.com>
```
Comment (Davidy22) [inline AUTHORS:1]: This file has tons of duplicate names, and why do we need this?

File: Makefile.in [hunk 3]
```diff
@@ -0,0 +1,870 @@
+# Makefile.in generated by automake 1.15 from Makefile.am.
```
Comment (Davidy22) [inline Makefile.in:1]: There's a bunch of these generated files in this pull request clogging up the changes, and these shouldn't be in a pull request anyway. I think these are already in the .gitignore, but if they aren't on your repo you need to add them back.

File: Pipfile [hunk 4]
```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [requires]
-python_version = "3"
+python_version = "3.10.0"
```
Comment (Davidy22) [inline Pipfile:7]: Don't do this, setting 3.10 in this file straight excludes earlier python versions.

File: config.guess [hunk 5]
```diff
@@ -0,0 +1,1462 @@
+#! /bin/sh
```
Comment (Davidy22) [inline config.guess:1]: Why is this in the repo?

File: guake/callbacks.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ def on_search_on_web(self, *args):
             query = clipboard.wait_for_text()
             query = quote_plus(query)
             if query:
-                search_url = f"https://www.google.com/search?q={query}&safe=off"
+                search_url = "https://www.google.com/search?q={!s}&safe=off".format(query)
```
Comment (Davidy22) [inline guake/callbacks.py:57]: What is this revert for? Have you not updated your repo from master yet?

**Review by Davidy22 — CHANGES_REQUESTED** [review]
I'm scanning the changes and there's an absolute ton. In addition to whatever I might flag, you need to write release notes files describing what you've changed yourself because there's no way I'm going to be able to write it for you when there's this much changed in one pull request. Also, this pull request fails CI right now, I think CI was broken last time you submitted changes but it's back now.

A good number of your functional commits may have been functionally duplicated since they were first written, so they'll either introduce duplicate code or do nothing. Ideally we don't want to add any of those to the commit log. For sanity and manageability, it'd probably be easier to process any changes if you just cherry picked commits that are still relevant into a new branch and made the pull request from there, instead of pushing this monolithic pull request.

**Comment by Davidy22** [comment]
I notice a commit's been added, but we haven't resolved CI failing or even merge conflicts. I'm not going to resolve merge conflicts for you, and I'm not going to pull generated files or random reverts caused by not updating. I'm going to close this pull request in a day if there's no verbal response, and if you can't reform this into just the fixes you're adding I'm going to close this so that you can cherry pick the functional changes into separate smaller pull requests.

File: po/ca.po (hunk 1 above)
Comment (zeusintuivo) [reply po/ca.po:1533]: I did rebase from master. I takes a lot of effort as well, to get Guake to a stable version. I feel the same that you guys keep maintaining a broken master, and that my branch has the good things that make a good guake. I am faced with the question of having guake prs_combined self-compiled or get the default /apt/dnf old 3.2 version with lack of real useful functions.

File: AUTHORS (hunk 2 above)
Comment (zeusintuivo) [reply AUTHORS:1]: This I did not understand, "duplicate" I thought this file is getting generated automatically noh?

File: Pipfile (hunk 4 above)
Comment (zeusintuivo) [reply Pipfile:7]: What would be a good version just "3" ... I can set it back. No preference here. As long as it works

File: config.guess (hunk 5 above)
Comment (zeusintuivo) [reply config.guess:1]: are these files automatically generated?

File: guake/callbacks.py (hunk 6 above)
Comment (zeusintuivo) [reply guake/callbacks.py:57]: I have updated, and have cherry picked to use format, because the "f" seems to fail, is not quite working well. I have no idea, why are we using it, when it fails to compile

**Comment by zeusintuivo** [comment]
I would rather get some feedback about how to polish the required chances to make guake an stable version in my request. I am aware might be missing points and things, that I am aiming to fix, and not break further. 
If anyone can take the time to compile it and give feedback, as I am doing also. That would be great.

**Comment by mlouielu** [comment]
I second with @Davidy22, PR should be split up into several meaningful PRs, we do not accept this kind of large addition/deletion PR.

We can help you if you want, as a first step, maybe you can help us elaborate your idea and write it with bullet points.

**Comment by zeusintuivo** [comment]
> I'm scanning the changes and there's an absolute ton. In addition to whatever I might flag, you need to write release notes files describing what you've changed yourself because there's no way I'm going to be able to write it for you when there's this much changed in one pull request. Also, this pull request fails CI right now, I think CI was broken last time you submitted changes but it's back now.
> 
> A good number of your functional commits may have been functionally duplicated since they were first written, so they'll either introduce duplicate code or do nothing. Ideally we don't want to add any of those to the commit log. For sanity and manageability, it'd probably be easier to process any changes if you just cherry picked commits that are still relevant into a new branch and made the pull request from there, instead of pushing this monolithic pull request.

What would be a good way of communicating all the chances? Seems to me that I started this branch about 3 years ago, and the owners of this project slept and did not nothing, while lots of push requests were getting requested. 
I felt the need to use this project, since I believe it has a lot of potential, and I took careful effort to make stable and make it compile and work in Ubuntu and Fedora. 
Then you guys awoke, and took some pull-requests here and there, but lots of fixes, got skipped and ignored, so what ever is in master, got better, but I can notice, lots of fixed fixes, did not. perhaps you did not see them. 
I trying to keep things like for example "click to open" is completely broken. 
Another thing I remember is the adding more filters to click.
A very annoying thing that is very broken is that the JSON file the saves the session, keeps getting deleted, I thought I have fixed that already,
These samples I had fixed, and they are not yet in master.
I also add a script to compile Guake. This I find with extreme pain, and this script makes it way easier to do. 
I another is the declared variables like "visible=true" which are also failing. I had upgraded and fixed, and I noticed that that also has made it to master. 
In conclusion, it is an attempt to make it better not batter over it. 
Now I have my doubts if my fixes remain, since, it is also very hard to keep tract of all the little changes, where I see, tha old broken code gets pushed and merged to master. Then I have to fix it again and again, over and over.

**Comment by mlouielu** [comment]
> I would rather get some feedback about how to polish the required chances to make guake an stable version in my request. I am aware might be missing points and things, that I am aiming to fix, and not break further. If anyone can take the time to compile it and give feedback, as I am doing also. That would be great.

For example, what do you mean `a "better" version of Guake`, is about code quality, new function, or any thing else?

We need to know what kind of progress, improvement or bugfix has finished in your PR, and at least separate them from this mega PR.

Also, most of your commit didn't work well, for example, in Guake, if feature branch want to follow up with main branch, we rebase it, not cherry-pick or merge it from main branch.

File: config.guess (hunk 5 above)
Comment (Davidy22) [reply config.guess:1]: This one doesn't say it's automatically generated, but I don't see a reason why you'd copy and paste this full file into the repo so I assume this was generated by a tool you used.

File: Pipfile (hunk 4 above)
Comment (Davidy22) [reply Pipfile:7]: 3  will make it not exclude people on older versions of python

File: AUTHORS (hunk 2 above)
Comment (Davidy22) [reply AUTHORS:1]: This file is not in master. I think a tool that you're using is generating this file, and it's done so quite recently because there's some newer names in this file. I don't know why we'd need this file. By duplicate names, I mean there's places where the same name shows up multiple times with different email addresses, like gaetan semet with a multitude of different email addresses.

File: Makefile.in (hunk 3 above)
Comment (Davidy22) [reply Makefile.in:1]: These makefiles won't be cleaned up by a rebase, these generated files were never in master to begin with so a rebase will never tell your branch to delete them. We could add some lines to the gitignore, or we can go through and clean out all the generated makefiles in your repo. A hefty amount of the junk in the changed files view is actually these files, so your aggregate changeset probably gets significantly easier to review after we don't need to scroll past all these identical files.

File: po/ca.po (hunk 1 above)
Comment (Davidy22) [reply po/ca.po:1533]: It may be significantly easier to create branches from Guake master, and cherry pick your changes into the branches. Try to keep it one feature per branch, and open a seperate PR for each feature so that we can check them all feasibly. It'll be easier to avoid merge conflicts too since you'll be starting from master.

File: guake/callbacks.py (hunk 6 above)
Comment (Davidy22) [reply guake/callbacks.py:57]: f strings aren't supported in python 3.5 or below, but 3.5 is also very very EOL

**Comment by Davidy22** [comment]
At the base level, there's CI that needs to pass. It's a bunch of fairly simple checks for code quality/that things still work, all the things with red crosses next to them should also have a button that says "Details" somewhere around them that'll tell you what needs  to be fixed, and that's just the automated checks.

The project should have been handed off earlier before PRs had a chance to languish, but can't undo the past. For now, if you have material improvements to make you'll need to fill in release notes file(s) describing what changed, because we can't just say in our release notes "3 years of misc changes." If you can get all the merge conflicts and random generated files all cleaned out of this I could consider a squash and merge, but I expect it's going to be much easier for you to just reform this by branching off Guake master and pulling relevant changes into a copy of current master.

**Comment by zeusintuivo** [comment]
I am going to close this and make smaller requests better
